tc/core/tc2halide.cc

```diff
@@ -593,8 +588,9 @@ void translateComprehension(
                                  << c.assignment()->range().text() << "\n";
   }
 
+  // Tag reductions as such
   if (c.assignment()->kind() != '=') {
-    reductions->push_back(f);
+    rhs = reductionUpdate(rhs);
```

Comment:
This is way simpler than the old code. The only thing that gives me pause is that it means the reductionUpdate intrinsics now appear in the function definitions, rather than just the lowered stmt. This may break internal Halide analysis passes that run over the functions (e.g. predicting the range of values they could take on to bound things like the access to f in f(g(idx))), factoring associative reductions). I don't think we're using any of that machinery right now though. If we ever send this DAG of Funcs down the regular Halide lowering path instead (e.g. to hit some exotic architecture), then we'll need to run a mutator on the Function DAG that strips these intrinsics out. 

For this PR, probably just add a comment:

Note: wrapping the function definitions in this intrinsic will hinder some of the analysis passes in Halide's standard lowering path. If we ever use that, or more pieces of that, we should revisit this.